esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -77,47 +78,63 @@ def _fix_dim_coordnames(cube):
         coord_type = iris.util.guess_coord_axis(coord)
+        try:
+            coord = cube.coord(axis=coord_type)
+        except iris.exceptions.CoordinateNotFoundError:
+            logger.warning(
+                'Multiple coordinates for axis %s. '
+                'This may be an error, specially for regular grids',
+                coord_type
+            )
+            continue
 
         if coord_type == 'T':
-            cube.coord(axis=coord_type).var_name = 'time'
-            cube.coord(axis=coord_type).attributes = {}
+            coord.var_name = 'time'
+            coord.attributes = {}
 
         if coord_type == 'X':
-            cube.coord(axis=coord_type).var_name = 'lon'
-            cube.coord(axis=coord_type).standard_name = 'longitude'
-            cube.coord(axis=coord_type).long_name = 'longitude coordinate'
-            cube.coord(axis=coord_type).units = Unit('degrees')
-            cube.coord(axis=coord_type).attributes = {}
+            coord.var_name = 'lon'
+            coord.standard_name = 'longitude'
+            coord.long_name = 'longitude coordinate'
+            coord.units = Unit('degrees')
+            coord.attributes = {}
 
         if coord_type == 'Y':
-            cube.coord(axis=coord_type).var_name = 'lat'
-            cube.coord(axis=coord_type).standard_name = 'latitude'
-            cube.coord(axis=coord_type).long_name = 'latitude coordinate'
-            cube.coord(axis=coord_type).units = Unit('degrees')
-            cube.coord(axis=coord_type).attributes = {}
+            coord.var_name = 'lat'
+            coord.standard_name = 'latitude'
+            coord.long_name = 'latitude coordinate'
+            coord.units = Unit('degrees')
+            coord.attributes = {}
 
         if coord_type == 'Z':
-            if cube.coord(axis=coord_type).var_name == 'depth':
-                cube.coord(axis=coord_type).standard_name = 'depth'
-                cube.coord(axis=coord_type).long_name = \
+            if coord.var_name == 'depth':
+                coord.standard_name = 'depth'
+                coord.long_name = \
                     'ocean depth coordinate'
-                cube.coord(axis=coord_type).var_name = 'lev'
-                cube.coord(axis=coord_type).attributes['positive'] = 'down'
-            if cube.coord(axis=coord_type).var_name == 'pressure':
-                cube.coord(axis=coord_type).standard_name = 'air_pressure'
-                cube.coord(axis=coord_type).long_name = 'pressure'
-                cube.coord(axis=coord_type).var_name = 'air_pressure'
-                cube.coord(axis=coord_type).attributes['positive'] = 'up'
+                coord.var_name = 'lev'
+                coord.attributes['positive'] = 'down'
+            if coord.var_name == 'pressure':
+                coord.standard_name = 'air_pressure'
+                coord.long_name = 'pressure'
+                coord.var_name = 'air_pressure'
+                coord.attributes['positive'] = 'up'
 
```

Comment:
OK this is cool, but changing `coord` will automatically change the cube as well? (I didn't know this is an in-place operation)

Reply:
Yes, the only difference with the old code is that I keep the object reference and do not look for it every time